.github/workflows/main.yml

```diff
@@ -0,0 +1,154 @@
+name: Main
+
+on:
+  push:
+    branches: [ master, develop, ci ]
+  pull_request:
+    branches: [ master, develop ]
+
+jobs:
+
+  build_and_test:
+      runs-on: ${{ matrix.os }}
+
+      strategy:
+        fail-fast: false
+
+        matrix:
+          os: [ubuntu-latest, macos-latest, windows-latest]
+          python-version: [3.9]
+          cmake_flags: ['', '-DUNITTESTS=ON -DCOVERAGE=ON']
+          cmake_flags_extra: ['', '-DENABLE_MKL_PARDISO=OFF', '-DDFLOAT=ON', '-DDLONG=OFF', '-DEMBEDDED=1',
+                              '-DEMBEDDED=2', '-DPROFILING=OFF', '-DCTRLC=OFF', '-DPRINTING=OFF']
+          exclude:
+            - cmake_flags: '-DUNITTESTS=ON -DCOVERAGE=ON'
+              cmake_flags_extra: '-DEMBEDDED=1'
+            - cmake_flags: '-DUNITTESTS=ON -DCOVERAGE=ON'
+              cmake_flags_extra: '-DEMBEDDED=2'
+
+      defaults:
+        run:
+          # Required when using an activated conda environment in steps
+          # See https://github.com/conda-incubator/setup-miniconda#IMPORTANT
+          shell: bash -l {0}
+
+      steps:
+        - uses: actions/checkout@v2
+          with:
+            lfs: false
+            submodules: true
+
+        - name: Set up conda
+          uses: conda-incubator/setup-miniconda@v2
+          with:
+            auto-update-conda: true
+            python-version: ${{ matrix.python-version }}
+
+        # -----------------
+        # OS-specific setup
+        # -----------------
+        - name: Setup (Linux)
```

Comment:
I think it might be better to put the CMake generators into the matrix environment along with the other CMake stuff (and since we probably want to add a Visual Studio generator at some point to test compiling on that):
```
include:
  - os: ubuntu-latest
    cmake_generator: "Unix Makefiles"
  - os: macos-latest
    cmake_generator: "Unix Makefiles"
  - os: windows-latest
    cmake_generator: "MinGW Makefiles"
```
and then the call to CMake in the build stage would be:
```
cmake -G ${{ matrix.cmake_generator }} -S . -B build ${{ matrix.cmake_flags }} ${{ matrix.cmake_flags_extra }}
```

Reply:
Done (needs quotes around the variable, as I discovered)